lib/pages/bucket_page/bucket_page.dart

```diff
@@ -66,81 +95,47 @@ class BucketPage extends StatelessWidget {
+                ? splittedPrefix[splittedPrefix.length - 2]
+                : splittedPrefix[0];
+            List<String> newPath = [...state.path];
+            newPath.add(directoryName);
+
+            if (directoryName.contains(state.filter)) {
+              directoriesWidgets.add(ListTile(
+                leading: const Icon(FluentIcons.folder_horizontal),
+                title: Text(directoryName.replaceAll('/', '')),
+                onPressed: () {
+                  context
+                      .read<BucketPageBloc>()
+                      .add(DirectoryAdded(path: newPath));
+                  context
+                      .read<BucketPageBloc>()
+                      .add(ObjectsRequested(prefix: newPath.join('/')));
+                },
+              ));
+            }
+          }
           List<ListTile>? objectsWidgets = [];
 
-          objects.forEach(
-            (object) {
-              String? objectName = object.key?.split('/').last;
-              if (objectName != '' &&
-                  objectName != null &&
-                  objectName.contains(state.filter)) {
-                objectsWidgets.add(ListTile(
-                  leading: FileIcon(
-                    objectName,
-                    size: 22,
-                  ),
-                  trailing: const Icon(FluentIcons.download),
-                  title: Text(objectName),
-                  onPressed: () async {
-                    var downloadDir =
-                        await path_provider.getDownloadsDirectory();
-                    String? selectedDirectory = await FilePicker.platform
-                        .getDirectoryPath(
-                            initialDirectory: downloadDir?.path ?? '/');
-                    if (selectedDirectory != null) {
-                      Minio minio = Minio(
-                          endPoint: connection.endpoint,
-                          accessKey: connection.accessKey,
-                          secretKey: connection.secretKey);
-
-                      await minio.fGetObject(
-                          bucket.name,
-                          object.key!,
-                          path_lib.join(selectedDirectory,
-                              path_lib.basename(object.key!)));
-
-                      displayInfoBar(context, builder: (context, close) {
-                        return InfoBar(
-                          title: const Text('File downloaded'),
-                          content: Text(object.key!),
-                          action: IconButton(
-                            icon: const Icon(FluentIcons.clear),
-                            onPressed: close,
-                          ),
-                          severity: InfoBarSeverity.success,
-                        );
-                      });
-                    }
-                  },
-                ));
-              }
-            },
-          );
+          for (var object in objects) {
```

Comment:
up

Reply:
dart fix did that

Reply:
i reverted that decision and disabled that check